llvm/lib/Target/X86/X86InstrAVX512.td

```diff
@@ -1447,6 +1447,17 @@ def : Pat<(vselect_mask VK8WM:$mask,
           (VBROADCASTI32X4Z256rmk VR256X:$src0, VK8WM:$mask, addr:$src)>;
 }
 
+let Predicates = [HasBF16] in {
```

Comment:
Why `HasBF16` there, the pattern didn't use AVX512BF16 instruction. Maybe use HasAVX512 like fp16's pattern?

Reply:
Because v[8,16,32]bf16 are legal only when AVX512BF16 or AVXNECONVERT features are ready.

Reply:
Got it. Thanks!